huaweicloud/services/aom/resource_hauweicloud_aom_prometheus_instance.go

```diff
@@ -0,0 +1,149 @@
+		},
+
+		Schema: map[string]*schema.Schema{
+			"region": {
+				Type:     schema.TypeString,
+				Optional: true,
+				Computed: true,
+				ForceNew: true,
+			},
+			"project_id": {
+				Type:     schema.TypeString,
+				Required: true,
+				ForceNew: true,
+				ValidateFunc: validation.All(
+					validation.StringLenBetween(1, 100),
+					validation.StringMatch(regexp.MustCompile(
+						"^[\u4e00-\u9fa5A-Za-z0-9]([\u4e00-\u9fa5-_A-Za-z0-9]*[\u4e00-\u9fa5A-Za-z0-9])?$"),
+						"The name can only consist of letters, digits, underscores (_),"+
+							" hyphens (-) and chinese characters, and it must start and end with letters,"+
+							" digits or chinese characters."),
+				),
+			},
+			"prom_for_cloud_service": {
+				Type:     schema.TypeList,
+				Optional: true,
+				ForceNew: true,
+				Elem: &schema.Resource{Schema: map[string]*schema.Schema{
+					"ces_metric_namespaces": {
+						Type:     schema.TypeList,
+						Optional: true,
+						Elem:     &schema.Schema{Type: schema.TypeString},
+					},
+				}},
+			},
+
+			"action": {
+				Type:     schema.TypeString,
+				Required: true,
+				ForceNew: true,
+			},
+		},
+	}
+}
+
+func resourcePrometheusInstanceRead(_ context.Context, d *schema.ResourceData, meta interface{}) diag.Diagnostics {
+	config := meta.(*config.Config)
+	client, dErr := httpclient_go.NewHttpClientGo(config)
+	if dErr != nil {
+		return dErr
+	}
+	client.WithMethod(httpclient_go.MethodGet).WithUrlWithoutEndpoint(config, "aom",
+		config.GetRegion(d), "v1/"+config.TenantID+"/prometheus-instances?action=prom_for_cloud_service")
+
+	resp, err := client.Do()
+	if err != nil {
+		return common.CheckDeletedDiag(d, err, "error retrieving AOM alarm rule")
+	}
+
+	mErr := &multierror.Error{}
+
+	defer resp.Body.Close()
+	body, err := io.ReadAll(resp.Body)
+	if err != nil {
+		mErr = multierror.Append(mErr, err)
+	}
+	rlt := &entity.PrometheusInstanceParams{}
+
+	err = json.Unmarshal(body, rlt)
+
+	if err != nil {
+		mErr = multierror.Append(mErr, err)
+	}
+	d.Set("prom_for_cloud_service", buildPrometheusInstanceMap(rlt))
+	d.Set("action", "prom_for_cloud_service")
+	d.Set("project_id", config.TenantID)
+	if err = mErr.ErrorOrNil(); err != nil {
+		return diag.Errorf("error getting AOM prometheus instance fields: %s", err)
+	}
+
+	return nil
+}
+
+func buildPrometheusInstanceMap(rlt *entity.PrometheusInstanceParams) []map[string]interface{} {
+	var m = make(map[string]interface{})
+	m["ces_metric_namespaces"] = rlt.PromForCloudService.CesMetricNamespaces
+	return []map[string]interface{}{m}
+}
+
+func resourcePrometheusInstancePatch(_ context.Context, d *schema.ResourceData, meta interface{}) diag.Diagnostics {
+	config := meta.(*config.Config)
+	client, dErr := httpclient_go.NewHttpClientGo(config)
+	if dErr != nil {
+		return dErr
+	}
+	var p = d.Get("prom_for_cloud_service").([]interface{})[0]
+	service := p.(map[string]interface{})["ces_metric_namespaces"].([]interface{})
```

Comment:
if `prom_for_cloud_service` and `ces_metric_namespaces` are optional, it will be panic if parameters are not specified.